File: src/taskpane/services/variableResolver.ts
import type { VariableMetadata } from "../types/variable";

export function resolveVariableValue(metadata: VariableMetadata): string {
  if (metadata.variableKey === "student_name" && metadata.studentId === 1) {
    return "John Doe";
  }

  if (metadata.variableKey === "course_title" && metadata.courseId === 101) {
    return "Introduction to Computer Science";
  }

  if (metadata.variableKey === "grade" && metadata.courseId === 101 && metadata.studentId === 1) {
    return "A";
  }

  return "[UNRESOLVED]";
}

File: scripts/dotnet/VariableResolver.cs
namespace ResolveVariables;

public static class VariableResolver
{
    public static string Resolve(VariableMetadata metadata) => metadata.VariableKey switch
    {
        "student_name" when metadata.StudentId == 1        => "John Doe",
        "course_title" when metadata.CourseId  == 101      => "Introduction to Computer Science",
        "grade"        when metadata.CourseId  == 101
                         && metadata.StudentId == 1        => "A",
        _                                                  => "[UNRESOLVED]"
    };
}
